openslides/core/static/js/core/site.js

```diff
@@ -815,15 +841,54 @@ angular.module('OpenSlidesApp.core.site', [
             $http.post('/rest/core/projector/1/control_view/', {"action": action, "direction": direction});
         };
         $scope.editCurrentSlide = function () {
-            $.each(Projector.get(1).elements, function(key, value) {
-                if (value.name == 'agenda/list-of-speakers') {
+            angular.forEach(Projector.get(1).elements, function(element) {
+                if (element.name == 'agenda/list-of-speakers') {
                     $state.go('agenda.item.detail', {id: value.id});
                 } else if (
-                    value.name != 'agenda/item-list' &&
-                    value.name != 'core/clock' &&
-                    value.name != 'core/countdown' &&
-                    value.name != 'core/message' ) {
-                    $state.go(value.name.replace('/', '.')+'.detail.update', {id: value.id});
+                    element.name != 'agenda/item-list' &&
+                    element.name != 'core/clock' &&
+                    element.name != 'core/countdown' &&
+                    element.name != 'core/message' &&
+                    element.name != 'core/speakeroverlay') {
+                    $state.go(element.name.replace('/', '.')+'.detail.update', {id: element.id});
+                }
+            });
+        };
+        //*** List of speakers overlay on slide***
+        $scope.speakeroverlay = function() {
+            angular.forEach(Projector.get(1).elements, function(element) {
+                if (element.name == 'core/speakeroverlay') {
+                    return element;
+                }
+            });
+        };
+        
+        $scope.speakeroverlaytoggle = function () {
+            var data = {};
+            if ($scope.speakeroverlay().visible) {
+                data[$scope.speakeroverlay().uuid] = { "visible": false };
+            } else {
+                data[$scope.speakeroverlay().uuid] = { "visible": true };
+            }
+            $http.post('/rest/core/projector/1/update_elements/', data);
+        };
+        $scope.goToListofSpeakers = function() {
+            angular.forEach(Projector.get(1).elements, function(element) {
+                if (element.name == 'motions/motion') {
+                    Motion.find(element.id).then(function(motion){
```

Comment:
Please check whether this can be done either by a resolve or by Motion.get(element.id).